src/Codegen/Constraints/ObjectBuilder.php

```diff
@@ -320,19 +320,19 @@ protected function getCheckMethodCode(
           ->addAssignment('$failed_any', false, HackBuilderValues::export())
           ->ensureEmptyLine()
           ->startForeachLoop('$patterns', '$pattern', '$constraint')
-          ->addAssignment('$matches', '[]', HackBuilderValues::literal())
+          ->addAssignment('$matches', 'varray[]', HackBuilderValues::literal())
```

Comment:
$matches ends up being used as the preg_match result. Passing in a varray is safe.

Reply:
However, in the codegen code $matches is never passed to preg_match, so it is technically unused. However I did not dare to remove it, since it might be used in code that is generated for different schema types.